packages/trader/src/App/Containers/Modals/__tests__/success-dialog.spec.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { render, screen, within } from '@testing-library/react';
+import SuccessDialog from '../success-dialog';
+import userEvent from '@testing-library/user-event';
+
+type TModal = React.FC<{
+    children: React.ReactNode;
+    is_open: boolean;
+    has_close_icon: boolean;
+    toggleModal: () => void;
+    title: string;
+}> & {
+    Body?: React.FC<{
+        children: React.ReactNode;
+    }>;
+    Footer?: React.FC<{
+        children: React.ReactNode;
+    }>;
+};
+
+jest.mock('@deriv/components', () => {
+    const original_module = jest.requireActual('@deriv/components');
+    const Icon = jest.fn(() => <div>Checkmark</div>);
+    const Modal: TModal = jest.fn(({ children, is_open, has_close_icon, toggleModal, title }) => {
+        if (is_open)
+            return (
+                <div data-testid='modal'>
+                    <h3>{title}</h3>
+                    {has_close_icon && <span onClick={toggleModal}>IcCross</span>}
+                    {children}
+                </div>
+            );
+
+        return null;
+    });
+    Modal.Body = jest.fn(({ children }) => <div>{children}</div>);
+    Modal.Footer = jest.fn(({ children }) => <div>{children}</div>);
+
+    return {
+        ...original_module,
+        Modal,
+        Icon,
+    };
+});
+
+describe('<SuccessDialog />', () => {
+    const modal_root_el = document.createElement('div');
```

Comment:
Just asking: did you write it in order to create a portal? If yes you cam do it by adding ⬇️ , because we try to avoid using non-testing library methods (like `createElement`, `setAttribute`, `appendChild`)
```
beforeAll(() => {
        (ReactDOM.createPortal as jest.Mock) = jest.fn(component => {
            return component;
        });
    });
```

Reply:
mmm actually i don't know. A lot of our tests seem to use this method for creating a modal. Just following the same convention.